### PR TITLE
Mapped CircaDB and treatment data to hetionet IDs

### PR DESCRIPTION
**Pull request by yhao-compbio**

### CODE REVIEW

File: explore/circadian-efficacy/data/disease_doid.tsv [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+myocardial injury	DOID:5844
+myocardial injury/cardiovascular thromboembolism	DOID:5844
+NSCLC	DOID:3910
+oral squamous cell carcinoma	DOID:0050866
+Osteoarthritis	DOID:8398
+painful bone metastases	NA
+prostate adenocarcinoma	DOID:10283
+Rheumatoid arthritis	DOID:7148
+skin allergy	NA
+soft tissue or bone sarcoma	NA
```
Comment (cgreene) [inline explore/circadian-efficacy/data/disease_doid.tsv]: Does DOID:1115 capture this?

**Comment by dhimmel** [comment]
Here are the primary commands to help you move 3530a39a1a500ee51ef05126d42be32a73a7b933 to a separate pull request:

```
# undo commit for other PR on this branch
git revert 3530a39a1a500ee51ef05126d42be32a73a7b933

# Checkout master, then checkout new branch for other PR
git cherry-pick 3530a39a1a500ee51ef05126d42be32a73a7b933
```

File: explore/circadian-efficacy/README.md [hunk 2]
```diff
@@ -10,3 +10,25 @@ Evaluating unsupervised DWPC p-values and path contributions to predict whether
 + [`downloads/aat8806_Data_file_S1.xlsx`](downloads/aat8806_Data_file_S1.xlsx) contains data from CircaDB, which provides tissue-specific rhythmical scores for human genes. 
   The data was downloaded from Data file S1 of [A database of tissue-specific rhythmically expressed human genes has potential applications in circadian medicine](https://doi.org/10.1126/scitranslmed.aat8806).
   The algorithm 'CYCLOPS', which was developed to score genes based on circadian rhythms, can be accessed from [CYCLOPS reveals human transcriptional rhythms in health and disease](https://doi.org/10.1073/pnas.1619320114).
+
++ [`downloads/drug_links.csv`](downloads/drug_links.csv) contains mapping between compound names and DrugBank IDs, as well as other external links.
+  The data was downloaded from [External Drug Links in Drugbank](https://www.drugbank.ca/releases/5-1-3/downloads/all-drug-links).
```
Comment (dhimmel) [inline explore/circadian-efficacy/README.md]: What do we use `drug_links.csv` for? I am worried this is not the right place to process or store drugbank cross-references.

File: explore/circadian-efficacy/README.md [hunk 3]
```diff
@@ -10,3 +10,25 @@ Evaluating unsupervised DWPC p-values and path contributions to predict whether
...
+
++ [`downloads/GTEx_Analysis_2016-01-15_v7_RNASeQCv1.1.8_gene_median_tpm.gct`](downloads/GTEx_Analysis_2016-01-15_v7_RNASeQCv1.1.8_gene_median_tpm.gct) contains median expression of all genes in 55 GTEx tissues.
+  The data was downloaded from [GTEx V7 RNA-seq data](https://storage.googleapis.com/gtex_analysis_v7/rna_seq_data/GTEx_Analysis_2016-01-15_v7_RNASeQCv1.1.8_gene_median_tpm.gct.gz)
```
Comment (dhimmel) [inline explore/circadian-efficacy/README.md]: This file is 13.8 MB. Let's compress it to make it smaller and cause it to be tracked via Git LFS. Either gzip or xz.

File: explore/circadian-efficacy/README.md [hunk 4]
```diff
@@ -10,3 +10,25 @@ Evaluating unsupervised DWPC p-values and path contributions to predict whether
...
+
+# Generated datasets
+
++ [`data/disease_doid.tsv`](data/disease_doid.tsv) contains manually curated mapping from therapeutic area to disease ontology ID.
+
++ [`data/tissue_uberon.tsv`](data/tissue_uberon.tsv) contains manually curated mapping from CircaDB tissue terms to exact Uberon tissue IDs, hetionet Uberon tissue IDs(closest tissue available in hetionet), and GTEx tissue names.
```
Comment (dhimmel) [inline explore/circadian-efficacy/README.md]: Shouldn't we also have a manual `compound_map.tsv`? For example, I noticed "Folinic Acid" does not get mapped to `DB00650`?

File: explore/circadian-efficacy/data_id_mapping.ipynb [hunk 5]
```diff
@@ -0,0 +1,176 @@
+{
```
Comment (dhimmel) [inline explore/circadian-efficacy/data_id_mapping.ipynb]: Let's always add a title markdown cell to our notebooks with a brief description of what they do.

File: explore/circadian-efficacy/data_id_mapping.ipynb [hunk 6]
```diff
@@ -0,0 +1,176 @@
...
+   "source": [
+    "# output ndarray that contains FDR, amplitude, expression of all genes measured\n",
+    "f1 = open('data/circa_db_mapped.tsv', \"w\")\n",
+    "\n",
+    "combine_names_string = '\\t'.join(cn for cn in combine_names)\n",
+    "f1.write('gene_id\\t%s\\n' % combine_names_string)\n",
+    "\n",
+    "for i in range(0, len(combine_array)):\n",
+    "    combine_string = '\\t'.join(format(ca,'.4f') for ca in combine_array[i])\n",
+    "    f1.write('%d\\t%s\\n' % (gene_list[i], combine_string))\n",
```
Comment (dhimmel) [inline explore/circadian-efficacy/data_id_mapping.ipynb]: Do not write CSVs / TSVs by manually. Instead use python's `csv` module or create a pandas.DataFrame and use df.to_csv.

File: explore/circadian-efficacy/data_id_mapping.ipynb [hunk 7]
```diff
@@ -0,0 +1,176 @@
...
+    "all_genes_ensg = []\n",
+    "\n",
+    "for gene in all_genes:\n",
+    "    gene_id = circa_db.index[circa_db['Entrez.ID'] == gene]\n",
+    "    if len(gene_id) == tissue_len:\n",
+    "        all_genes_ensg.append(circa_db.iloc[gene_id[0],1])\n",
+    "        gene_list.append(gene)\n",
+    "        \n",
+    "        gene_fdr = np.array(circa_db.iloc[gene_id,4])\n",
+    "        gene_amp = np.array(circa_db.iloc[gene_id,6])\n",
```
Comment (dhimmel) [inline explore/circadian-efficacy/data_id_mapping.ipynb]: Why are we accessing values in a dataframe by integer index? This is usually very brittle.

File: explore/circadian-efficacy/data_id_mapping.ipynb (hunk 5 above)
Comment (dhimmel) [reply explore/circadian-efficacy/data_id_mapping.ipynb]: Also can you show the first two or three rows of each dataframe (`df.head(2)`) for all important dfs in the notebook. One of the best parts of notebooks is that they can show output interspersed with code. Without this, it is hard to understand the notebook.

File: explore/circadian-efficacy/data/tissue_uberon.tsv [hunk 8]
```diff
@@ -0,0 +1,14 @@
+Tissue	Exact.Uberon.code	Exact.Uberon.term	Hetionet.Uberon.code	Hetionet.Uberon.term	GTEx.name
```
Comment (dhimmel) [inline explore/circadian-efficacy/data/tissue_uberon.tsv]: in general, we should use all lowercase underscore separated column names like `exact_uberon_term`. The exception is if we have inherited a column name from an input resource.

should `Tissue` be `circadb_tissue` for clarity?

File: explore/circadian-efficacy/data_id_mapping.ipynb [hunk 9]
```diff
@@ -0,0 +1,987 @@
...
+      ]
+     },
+     "execution_count": 3,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "# read in Drugbank ID map\n",
+    "drugbank = pd.read_csv('https://raw.githubusercontent.com/yhao-compbio/drugbank/gh-pages/data/drugbank.tsv', sep = '\\t', header = 0)\n",
```
Comment (dhimmel) [inline explore/circadian-efficacy/data_id_mapping.ipynb]: Change URL to:

https://github.com/dhimmel/drugbank/raw/6b9ae386d6ba4a0eca2d66d4b0337a6e90fe81f4/data/drugbank.tsv

Note the use of commit hash versioning (press `y` hotkey on github). Also see use of `raw` which you can switch to `blob` to get back to HTML view.

Switch any other links to GitHub-hosted datasets in this PR to use this format.

**Comment by dhimmel** [comment]
Can you also change the conda installation in `.travis.yml` to use `Miniconda2-4.6.14-Linux-x86_64.sh` to fix the following [build error](https://travis-ci.org/greenelab/hetmech/builds/533978992#L556)

```
$ conda env create --quiet --file environment.yml
Solving environment: ...working... failed
CondaValueError: Malformed version string '~': invalid character(s).
```

https://stackoverflow.com/a/56116249/4651668

**Pull request merged**